src/atomate2/lobster/schemas.py

```diff
@@ -621,21 +750,268 @@ def from_directory(
+                                    if hasattr(
+                                        value, "as_dict"
+                                    ):  # check if item has a `as_dict` method
+                                        # (i.e. it is a pymatgen object)
+                                        data["cohp_plot_data"][key] = value.as_dict()
+                                data_new = {attribute: data}
+                                _replace_inf_values(data_new)
+                                json.dump(data_new, f)
+                                if (
+                                    attribute != list(doc_here.__fields__.keys())[-1]
+                                ):  # add comma separator between two dicts
+                                    f.write(",")
+                            else:
+                                if attribute == "builder_meta":
+                                    builder_meta = doc_here.__getattribute__(
+                                        attribute
+                                    ).dict()
+                                    builder_meta["build_date"] = str(
+                                        doc.builder_meta.build_date.utcnow()
+                                    )
+                                    data = {attribute: builder_meta}
+                                else:
+                                    data = {
+                                        attribute: doc_here.__getattribute__(
+                                            attribute
+                                        ).dict()
+                                    }
+                                json.dump(data, f)
+                                if (
+                                    attribute != list(doc_here.__fields__.keys())[-1]
+                                ):  # add comma separator between two dicts
+                                    f.write(",")
+                        elif hasattr(doc_here.__getattribute__(attribute), "as_dict"):
+                            data = {
+                                attribute: doc_here.__getattribute__(
+                                    attribute
+                                ).as_dict()
+                            }
+                            json.dump(data, f)
+                            if (
+                                attribute != list(doc_here.__fields__.keys())[-1]
+                            ):  # add comma separator between two dicts
+                                f.write(",")
+                        else:
+                            data = {attribute: doc_here.__getattribute__(attribute)}
+                            json.dump(data, f)
+                            if (
+                                attribute != list(doc_here.__fields__.keys())[-1]
+                            ):  # add comma separator between two dicts
+                                f.write(",")
+                f.write("]")
```

Comment:
What is going on here? Can't you just use `jsanitize` to get a dict representation of the data and then use `dumpfn`?

Reply:
(Should have catched this. Sorry.)

Reply:
Here as well, I wrote this function as most of the json saving libraries tend to create Python representation of the entire taskdoc in the memory which results in too much memory usage. The way I am m writing the JSONS, I am able to avoid excessive memory usage (writing the file incrementally by converting one field at a time to dict representation).

Maybe I am overlooking something here. Any suggestions are welcome

Reply:
Out of interest, how big are the generated files?